### PR TITLE
Create a separate branch and move the next project to the root

### PR DESCRIPTION
**Pull request by MuhammadZeeshanAshraf**
Create a separate branch and move the next project to the root

### CODE REVIEW

**Review by saadpasta — COMMENTED** [review]
@MuhammadZeeshanAshraf There are three open PR for the same issue can you close the one which are not needed

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,32 +1,23 @@
...
+        uses: amondnet/vercel-action@v20
         with:
-          GITHUB_TOKEN : ${{ secrets.GITHUB_TOKEN }} # This is provided by GitHub.
-          BRANCH: gh-pages # The branch the action should deploy to.
-          FOLDER: build # The folder the action should deploy.
+          BRANCH: main-next
+          vercel-token: ${{ secrets.VERCEL_TOKEN }} # Required
+          github-token: ${{ secrets.GITHUB_TOKEN }} #Optional
+          vercel-org-id: ${{ secrets.ORG_ID}} #Required
+          vercel-project-id: ${{ secrets.PROJECT_ID}} #Required
```
Comment (saadpasta) [inline .github/workflows/deploy.yml:23]: Added the secret to the project setting 🙏  Hopefully this should make it work

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run lint
```
Comment (saadpasta) [inline .husky/pre-commit:4]: Let's remove the husky. We can add the lint rule to our CI pipeline and fail the CI if the lint fails

File: Dockerfile [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-RUN npm audit fix
-
-# Bundle app source
-COPY . /app
-
-# Make port 3000 available to the world outside this container
-EXPOSE 3000
-
-# Run app.js when the container launches
-CMD ["npm", "start"]
```
Comment (saadpasta) [inline Dockerfile:29]: We might need docker file for our Nextjs project also but we can ditch it now

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -1,32 +1,23 @@
...
-  push:
-    branches:
-      - master
-  schedule:
-    - cron: "0 12 * * 1" # see https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events
+name: deploy nexturl to vercel
+on: [push, pull_request]
 jobs:
-  build-and-deploy:
+  vercel:
```
Comment (saadpasta) [inline .github/workflows/deploy.yml:4]: Let's rename this to build and deploy to vercel

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf CI also failing I already added all the token that were required anything else that we might need

**Comment by MuhammadZeeshanAshraf** [comment]
> @MuhammadZeeshanAshraf There are three open PR for the same issue can you close the one which are not needed

Extra PR closed

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf Have you fixed the above comments 🙏

**Comment by saadpasta** [comment]
Look there are other people having errors for this. Have a look at this [blog](https://dev.to/chuddyjoachim/nextjs-ci-cd-on-vercel-with-github-actions-7g7) comments

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf Please try using action from [here](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf Thanks a lot for contributing I already created a Next project and merged but I would still add you as a contributor for your efforts man. Thanks ❤️

**Pull request closed without merging**